### ISSUE TITLE
All values of tensor become Zero and training stops

### ISSUE THREAD

**Issue by mahavird** [andy-yun/pytorch-0.4-yolov3#44]
Hi @andy-yun ,

Thanks for your amazing work.

But I am facing some issue while using your code fro training a model using pre-trained yolov3.weights and VOC dataset.

The models gets loaded successfully while training but stops abruptly with all the tensor values zero.

Can you please help me out in debugging the cause ?

I have below attached the images for the same.

![train_1](https://user-images.githubusercontent.com/9322397/50590379-c560d000-0eb0-11e9-8eff-f5e5674edacc.png)
![screen_2](https://user-images.githubusercontent.com/9322397/50590385-cd207480-0eb0-11e9-8d25-930dd544133b.png)
![screen_4](https://user-images.githubusercontent.com/9322397/50590390-d0b3fb80-0eb0-11e9-8f46-08ce1122e005.png)

With Regards,
Mahavir

**Comment by pjwang317** [comment]
I think better to change the learning rate smaller

**Comment by mahavird** [comment]
Hi @winniewangji ,

Thanks for your feedback, I tried doing that . 

I reduced the learning rate from 0.001 to 0.0001 , but still the training stops after some time with a Zero matrix.
Shall I reduce the learning rate further ?

**Comment by andy-yun** [comment]
@mahavird It may be occurred you use  batch=1 and subdivitions=1 in your yolov3-voc.cfg file.
You can increase batch=16,32,64 and subdivitions=8,16,32,64 along to your environment.
batch=1 and subdivitions=1 is recommended for testing.
Even though you increase batch and subdivisions, nan error might be occurred.
In that case, you can avoid nan error by skipping the loss_conf if nan valus is found.
The avoiding technique seems heuristic but, you can avoid nan error.
The following is the sample code:

if math.isnan(loss_conf.item()):
   loss = loss_coord + loss_cls
else:
  loss = loss_coord + loss_conf+loss_cls
